From 018aa19c25c3f1d033b652d12f6244785c68f607 Mon Sep 17 00:00:00 2001
From: Joe Zheng <joe.zheng@intel.com>
Date: Wed, 17 Jul 2019 13:13:03 +0800
Subject: [PATCH] Restore rootfs context for SELinux enabling

Restore the context for rootfs when we are in the kernel domain

AIC does not use ramdisk and docker will drop xattrs when build image
so that is the reason we need to restorecon here

Change-Id: Ia83b394c2380f1679811735d979211d8108116cd
Topic: aic.selinux
Tracked-On: ACP-649
(cherry picked from commit d40a24cdb5d9c14518f06d04f686c20a728ce41e)
---
 init/init.cpp | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/init/init.cpp b/init/init.cpp
index dfff85fe7..d491119bc 100644
--- a/init/init.cpp
+++ b/init/init.cpp
@@ -627,6 +627,15 @@ int main(int argc, char** argv) {
         SelinuxSetupKernelLogging();
         SelinuxInitialize();
 
+        // restore the context for rootfs when we are in the kernel domain
+        // AIC does not use ramdisk and docker will drop xattrs when build image
+        // so that is the reason we need to restorecon here
+        LOG(INFO) << "Running restorecon for rootfs...";
+        selinux_android_restorecon("/", SELINUX_ANDROID_RESTORECON_RECURSE);
+        selinux_android_restorecon("/ipc", SELINUX_ANDROID_RESTORECON_RECURSE|SELINUX_ANDROID_RESTORECON_CROSS_FILESYSTEMS);
+        selinux_android_restorecon("/proc/cmdline", 0);
+        LOG(INFO) << "Running restorecon for rootfs done";
+
         // We're in the kernel domain, so re-exec init to transition to the init domain now
         // that the SELinux policy has been loaded.
         if (selinux_android_restorecon("/init", 0) == -1) {
-- 
2.21.0

